src/app/checkout/cart-summary/cart-summary.spec.js

```diff
@@ -31,9 +31,9 @@ describe('checkout', function () {
 
     describe('onSubmit', () => {
       it('should emit an event', () => {
-        jest.spyOn(componentInstance.$rootScope, '$emit').mockImplementation(() => {})
-        self.controller.onSubmit(componentInstance)
-        expect(componentInstance.$rootScope.$emit).toHaveBeenCalledWith(submitOrderEvent)
+        jest.spyOn(self.controller.$rootScope, '$emit').mockImplementation(() => {})
```

Comment:
The more correct way to do this is to set up a `$rootScope.$on` listener for the event and make sure that the listener is called. That better captures the higher-level intent of the test case.